src/ray/gcs/gcs_server/gcs_placement_group_manager.h

```diff
@@ -141,31 +155,55 @@ class GcsPlacementGroupManager : public rpc::PlacementGroupInfoHandler {
   void OnPlacementGroupCreationSuccess(
       const std::shared_ptr<GcsPlacementGroup> &placement_group);
 
+  /// TODO-SANG Fill it up.
+  void RemovePlacementGroup(const PlacementGroupID &placement_group_id,
+                            StatusCallback on_placement_group_removed);
+
  private:
   /// Try to create placement group after a short time.
   void RetryCreatingPlacementGroup();
 
+  /// Mark the manager that there's a placement group scheduling going on.
+  void MarkSchedulingStarted(const PlacementGroupID placement_group_id);
+
+  /// Mark the manager that there's no more placement group scheduling going on.
+  void MarkSchedulingDone();
+
+  /// Check if the placement group of a given id is scheduling.
+  bool IsSchedulingInProgress(const PlacementGroupID &placement_group_id);
+
+  /// Check if there's any placement group scheduling going on.
+  bool IsSchedulingInProgress();
+
   /// The io loop that is used to delay execution of tasks (e.g.,
   /// execute_after).
   boost::asio::io_context &io_context_;
 
   /// Callback of placement_group registration requests that are not yet flushed.
-  absl::flat_hash_map<PlacementGroupID, EmptyCallback>
+  absl::flat_hash_map<PlacementGroupID, StatusCallback>
       placement_group_to_register_callback_;
 
   /// All registered placement_groups (pending placement_groups are also included).
   absl::flat_hash_map<PlacementGroupID, std::shared_ptr<GcsPlacementGroup>>
       registered_placement_groups_;
+
+  /// All created placement_groups.
+  absl::flat_hash_map<PlacementGroupID, std::shared_ptr<GcsPlacementGroup>>
+      created_placement_group_;
+
   /// The pending placement_groups which will not be scheduled until there's a resource
   /// change.
   std::deque<std::shared_ptr<GcsPlacementGroup>> pending_placement_groups_;
+
   /// The scheduler to schedule all registered placement_groups.
   std::shared_ptr<gcs::GcsPlacementGroupSchedulerInterface>
       gcs_placement_group_scheduler_;
+
   /// Used to update placement group information upon creation, deletion, etc.
   std::shared_ptr<gcs::GcsTableStorage> gcs_table_storage_;
+
   /// If a placement group is creating
-  bool is_creating_ = false;
+  SchedulingProgress scheduling_progress_;
```

Comment:
Why not just keep PlacementGroupID here and set to Nil if not creating?

Reply:
Btw, I kept methods such as `MarkSchedulingStarted`. (which are all one liner). If you think it is unnecessary, I will remove it, so lmk.